Automattic/wp-calypso
# Fix plugins-browser-list placeholders

#### Changes proposed in this Pull Request

* Refactored the logic of showing placeholders to cover all cases correctly
* Added a required prop to the test mock

#### Testing instructions



* Go to Calypso live
* Click on the "See All" button on the top right of the "Popular" or "New" sections and do a sanity check, the initial render should show the placeholder as before
* Scroll down and validate that the placeholders are being appended to the plugins list
* Search for a plugin
* Do a sanity check, the initial render should show the placeholder as before
* Navigate to other pages with the pagination control, and validate that the placeholders are displayed in place of the whole list

| Event, `props` | Before | After |
| ----- | ------ | ----- |
| Initial render (works as before), `plugins = [], showPlaceholders = false, paginated = false` | ![calypso localhost_3000_plugins_popular_gcsecseyfree wordpress com](https://user-images.githubusercontent.com/11555574/137014719-19e44b4f-e395-443d-a9e7-0aeaf37ecb1b.png) | ![calypso localhost_3000_plugins_popular_gcsecseyfree wordpress com](https://user-images.githubusercontent.com/11555574/137014719-19e44b4f-e395-443d-a9e7-0aeaf37ecb1b.png) |
| Initial render (works as before), `plugins = [], showPlaceholders = false, paginated = true` | ![calypso localhost_3000_plugins_popular_gcsecseyfree wordpress com_s=woocommerce](https://user-images.githubusercontent.com/11555574/137014863-6778857f-399a-421f-834e-881070810917.png) |  ![calypso localhost_3000_plugins_popular_gcsecseyfree wordpress com_s=woocommerce](https://user-images.githubusercontent.com/11555574/137014863-6778857f-399a-421f-834e-881070810917.png) |
| Requesting next page, `plugins = [ {...}, {...}, {...} ], showPlaceholders = true, paginated = true` | ![wordpress com_plugins_gcsecseyfree wordpress com_s=woocommerce (1)](https://user-images.githubusercontent.com/11555574/137015423-bcdccc16-be24-446c-a2ed-1cd58bdb1d0f.png) |  ![calypso localhost_3000_plugins_popular_gcsecseyfree wordpress com_s=woocommerce (1)](https://user-images.githubusercontent.com/11555574/137015347-7849c41e-5c46-45f7-843d-84e1ad3f4573.png) |
| Requesting next page, `plugins = [ {...}, {...}, {...} ], showPlaceholders = true, paginated = false` | ![wordpress com_plugins_gcsecseyfree wordpress com_s=woocommerce (2)](https://user-images.githubusercontent.com/11555574/137015624-e3da8de1-6195-42f7-b1f4-ca1644cae44d.png) | ![calypso localhost_3000_plugins_popular_gcsecseyfree wordpress com (1)](https://user-images.githubusercontent.com/11555574/137016045-27c0309b-cd75-467a-b9e4-e246504fe609.png) |



Fixes #56793
